src/geometry/manifold/tangent_bundle.lean

```diff
@@ -107,38 +108,40 @@ structure basic_smooth_bundle_core {𝕜 : Type*} [nondiscrete_normed_field 𝕜
   ((I '' (i.1.symm.trans j.1).source) ×ˢ (univ : set F)))
 
 /-- The trivial basic smooth bundle core, in which all the changes of coordinates are the
-identity. -/
-def trivial_basic_smooth_bundle_core {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+identity.
+
+TODO: Update -/
+def trivial_basic_smooth_vector_bundle_core {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
 {E : Type*} [normed_group E] [normed_space 𝕜 E]
 {H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
 (M : Type*) [topological_space M] [charted_space H M] [smooth_manifold_with_corners I M]
-(F : Type*) [normed_group F] [normed_space 𝕜 F] : basic_smooth_bundle_core I M F :=
-{ coord_change := λ i j x v, v,
+(F : Type*) [normed_group F] [normed_space 𝕜 F] : basic_smooth_vector_bundle_core I M F :=
+{ coord_change := λ i j x, linear_map.id,
   coord_change_self := λ i x hx v, rfl,
   coord_change_comp := λ i j k x hx v, rfl,
   coord_change_smooth := λ i j, times_cont_diff_snd.times_cont_diff_on }
 
-namespace basic_smooth_bundle_core
+namespace basic_smooth_vector_bundle_core
```

Comment:
It looks like this change of name, together with the renaming of `to_topological_fiber_bundle_core` to `to_topological_vector_bundle_core` contribute substantially to the diff in this PR.

Would it be possible to split this PR so that the proposed name changes happen first and then we can review the proposed new material second?

Reply:
The name changes reflect the changes in the content of the code. What is implemented now is a fiber bundle and what is implemented in the PR is a vector bundle. If I do as you say after the first PR a fiber bundle will be called a vector bundle. If for some reason the later changes are not approved or it takes a long time to approve them, is it fine to have this name discrepancy in Mathlib for a while?

Does it help if I point out with comments where the code actually changes apart from renamings and adding lemmas to simps? I believe there are very few points where the code properly changes

Reply:
I see, sorry I should have looked more closely.

Does this not mean that we are losing `basic_smooth_bundle_core` then?

Reply:
It does but I guess @sgouezel is ok with it? The idea is that this construction should be useful precisely to build the tangent bundle whereas a more general `smooth_fiber_bundle_core` and `smooth_vector_bundle_core` should be used in other situations. If `smooth_fiber_bundle_core` will be needed again it will be easy to bring it back in by all means!